### PR TITLE
Fix pools table filter

### PR DESCRIPTION
**Pull request by plehocky**
| Q                        | A 
| ------------------------ | ---
| Fixed Issue?             | `FD-465` 
| Tests Added + Pass?      | Yes
| Any Dependency Changes?  | No

https://frinxhelpdesk.atlassian.net/browse/FD-465?atlOrigin=eyJpIjoiNmI4OGYyZDA5MmFhNGRhNmE4MWYxNTE5MmU5Yjk1ZjQiLCJwIjoiaiJ9

Filtering by name was not working unless you opened pool config page and returned with browser back button wich had it cached. Added "items" dependency to useEffect of use-minisearch hook that fixed the issue.

Also fixed filtering by resource type. When selecting for example ipv4, search results were correct but when switching back to default search option no results were listed

### CODE REVIEW

File: packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ const PoolsPage: VoidFunctionComponent = () => {
     return <div>{error?.message}</div>;
   }
 
-  const isSelectedResourceTypeEmpty = selectedResourceType == null || selectedResourceType.trim().length === 0;
+  const isSelectedResourceTypeEmpty = selectedResourceType === 'Select resource type to filter';
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx:128]: why should you use specific string when we are sending empty value. try to use that empty value

**Review by MarcoMruz — CHANGES_REQUESTED** [review]

File: packages/frinx-resource-manager/src/components/search-filter-pools-bar.tsx [hunk 2]
```diff
@@ -36,11 +36,14 @@ const SearchFilterPoolsBar: VoidFunctionComponent<Props> = ({
           <Select
             data-cy="select-resource-type"
             value={selectedResourceType}
-            onChange={(e) => setSelectedResourceType && setSelectedResourceType(e.target.value)}
+            onChange={(e) =>
+              setSelectedResourceType &&
+              setSelectedResourceType(e.target.value === 'Select resource type to filter' ? '' : e.target.value)
+            }
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/components/search-filter-pools-bar.tsx:42]: do not use specific string value... set empty "" string to the default option in select and use the previous code for handling onChange event `onChange={(e) => setSelectedResourceType && setSelectedResourceType(e.target.value)}`

remove from the default option the onChange event handler - it is unnecessary

File: packages/frinx-resource-manager/src/components/search-filter-pools-bar.tsx [hunk 3]
```diff
@@ -36,11 +36,14 @@ const SearchFilterPoolsBar: VoidFunctionComponent<Props> = ({
...
             variant="outline"
             bgColor="white"
           >
-            <option onClick={() => setSelectedResourceType && setSelectedResourceType('')}>
+            <option onChange={() => setSelectedResourceType && setSelectedResourceType('')}>
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/components/search-filter-pools-bar.tsx:46]: remove this onChange and use value="" as mentioned in the previous comment

File: packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx [hunk 4]
```diff
@@ -125,7 +125,7 @@ const PoolsPage: VoidFunctionComponent = () => {
     return <div>{error?.message}</div>;
   }
 
-  const isSelectedResourceTypeEmpty = selectedResourceType == null || selectedResourceType.trim().length === 0;
+  const isSelectedResourceTypeEmpty = selectedResourceType === null || selectedResourceType.trim().length === 0;
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx:128]: use just ==, so that we are able to handle also undefined values

**Review by MarcoMruz — CHANGES_REQUESTED** [review]

**Review by MarcoMruz — APPROVED** [review]

**Review by soson — APPROVED** [review]

**Pull request merged**